src/TypeScript/EslintBridgeClient/EslintBridgeHttpWrapper.cs

```diff
@@ -0,0 +1,96 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
+{
+    internal interface IEslintBridgeHttpWrapper : IDisposable
+    {
+        Task<string> PostAsync(string serverEndpoint, object request = null);
+    }
+
+    [Export(typeof(IEslintBridgeHttpWrapper))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class EslintBridgeHttpWrapper : IEslintBridgeHttpWrapper
```

Comment:
This logic is common to all the API calls so I decided to place it in a dedicated class. It makes it easier to test the caller (https://github.com/SonarSource/sonarlint-visualstudio/pull/2258)